pkg/provider/kubernetes/gateway/kubernetes.go

```diff
@@ -966,10 +1366,10 @@ func throttleEvents(ctx context.Context, throttleDuration time.Duration, pool *s
 	return eventsChanBuffered
 }
 
-func isInternalService(forwardTo v1alpha1.HTTPRouteForwardTo) bool {
-	return forwardTo.ServiceName == nil &&
-		forwardTo.BackendRef != nil &&
-		forwardTo.BackendRef.Kind == traefikServiceKind &&
-		forwardTo.BackendRef.Group == traefikServiceGroupName &&
-		strings.HasSuffix(forwardTo.BackendRef.Name, "@internal")
+func isInternalService(serviceName *string, backendRef *v1alpha1.LocalObjectReference) bool {
```

Comment:
I do not understand that change. Why would we make a more complicated signature given that serviceName and backendRef are both fields of the same object?